app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/chart/BeaconChartFragment.java

```diff
@@ -39,14 +46,20 @@ public BeaconChartFragment() {
 
             @Override
             public boolean matches(IBeacon beacon) {
-                if (legacyUuid.equals(beacon.getProximityUuid())) {
-                    return true;
-                }
-                if (indoorPositioningUuid.equals(beacon.getProximityUuid())) {
-                    return true;
-                }
-                if (gateDetectionUuid.equals(beacon.getProximityUuid())) {
-                    return true;
+                if (filterView) {
```

Comment:
Create a different filter depending on `filterView`. You don't need to hold the `UUID`s in that case and can avoid the repeated evaluations.